src/Microsoft.Sbom.Api/Utils/Comparer/SbomEqualityComparer.cs

```diff
@@ -0,0 +1,291 @@
+    {
+        if (spdx22Files.Count != spdx30Files.Count)
+        {
+            return false;
+        }
+
+        var spdx22InternalSbomFileInfos = ConvertToSbomFiles(spdx22Files);
+        var spdx30InternalSbomFileInfos = ConvertToSbomFiles(spdx30Files, spdx30Elements, relationships);
+
+        return spdx22InternalSbomFileInfos.SetEquals(spdx30InternalSbomFileInfos);
+    }
+
+    internal bool CheckPackages(List<SPDXPackage> spdx22Packages, List<Package> spdx30Packages, List<Element> spdx30Elements, List<Relationship> relationships)
+    {
+        if (spdx22Packages.Count != spdx30Packages.Count)
+        {
+            return false;
+        }
+
+        var spdx22InternalSbomPackages = ConvertToSbomPackages(spdx22Packages);
+        var spdx30InternalSbomPackages = ConvertToSbomPackages(spdx30Packages, spdx30Elements, relationships);
+
+        return spdx22InternalSbomPackages.SetEquals(spdx30InternalSbomPackages);
+    }
+
+    internal bool CheckRelationships(List<SPDXRelationship> spdx22Relationships, List<Relationship> spdx30Relationships)
+    {
+        // Filtering out relationships that describe license information since SPDX 2.2 does not support these relationship types.
+        // The license relationships are linked to files and packages when comparing the two SBOMs in the CheckFiles and CheckPackages methods.
+        var spdx30RelationshipsNotRelatedtoLicenses = spdx30Relationships.Where(
+            element => element.RelationshipType != RelationshipType.HAS_CONCLUDED_LICENSE
+            && element.RelationshipType != RelationshipType.HAS_DECLARED_LICENSE)
+            .ToList();
+
+        if (spdx22Relationships.Count != spdx30RelationshipsNotRelatedtoLicenses.Count)
+        {
+            return false;
+        }
+
+        var spdx22InternalRelationships = ConvertToRelationships(spdx22Relationships);
+        var spdx30InternalRelationships = ConvertToRelationships(spdx30RelationshipsNotRelatedtoLicenses);
+
+        return spdx22InternalRelationships.SetEquals(spdx30InternalRelationships);
+    }
+
+    internal bool CheckExternalDocRefs(List<SpdxExternalDocumentReference> spdx22ExternalDocumentRefs, List<ExternalMap> spdx30ExternalDocumentRefs)
+    {
+        if (spdx22ExternalDocumentRefs.Count != spdx30ExternalDocumentRefs.Count)
+        {
+            return false;
+        }
+
+        var spdx22InternalExternalDocRefs = ConvertToSbomReferences(spdx22ExternalDocumentRefs);
+        var spdx30InternalExternalDocRefs = ConvertToSbomReferences(spdx30ExternalDocumentRefs);
+
+        return spdx22InternalExternalDocRefs.SetEquals(spdx30InternalExternalDocRefs);
+    }
+
+    private HashSet<SbomFile> ConvertToSbomFiles(List<SPDXFile> files)
+    {
+        var sbomFiles = new HashSet<SbomFile>(sbomFileComparer);
+
+        foreach (var file in files)
+        {
+            sbomFiles.Add(file.ToSbomFile());
```

Comment:
What is the expectation on the case-sensitivity of these `HashSet` objects? I know that we have some code that uses `IOSUtils.IsCaseSensitiveOS` to handle casing differently on Windows vs Linux or macos